### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by bn-maksymmaliuk**

### CODE REVIEW

**Review by Zelinskyi-Serhii — APPROVED** [review]
No questions.

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,60 @@
...
-// create SortOrder type
-export type SortOrder;
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const studentsCopy = [...students];
+
+  return studentsCopy.sort((currentStudent, previousStudent): number => {
```
Comment (sergii-nosachenko) [inline src/sortStudents.ts:34]:
```suggestion
  return [...students].sort((currentStudent, previousStudent): number => {
```
Useless variable

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,60 @@
...
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return order === SortOrder.ASC
+          ? currentStudent[sortBy].localeCompare(previousStudent[sortBy])
+          : previousStudent[sortBy].localeCompare(currentStudent[sortBy]);
```
Comment (sergii-nosachenko) [inline src/sortStudents.ts:40]: DRY: you don't need to have both comparisons. Just make one and multiply the result by -1 in the case of `desc` order.
Fix it everywhere.

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,60 @@
...
+          ? calculateAverageGrade(currentStudent)
+            - calculateAverageGrade(previousStudent)
 
+          : calculateAverageGrade(previousStudent)
+            - calculateAverageGrade(currentStudent);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+      default:
+        throw new Error('Sort types are invalid');
```
Comment (sergii-nosachenko) [inline src/sortStudents.ts:57]: Use the template string to display the faulty `sortBy` value in the message.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,16 +1,53 @@
...
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students].sort((currtStudent, prevStudent): number => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return currtStudent[sortBy].localeCompare(prevStudent[sortBy])
+          * (order === SortOrder.ASC ? 1 : -1);
```
Comment (sergii-nosachenko) [inline src/sortStudents.ts:37]: DRY. Create a variable to hold this value.

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/sortStudents.ts [hunk 5]
```diff
@@ -1,16 +1,55 @@
...
+  return [...students].sort((currtStudent, prevStudent): number => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return currtStudent[sortBy].localeCompare(prevStudent[sortBy])
+          * orderCondition;
+
+      case SortType.Age:
+      case SortType.Married:
+        return (+(currtStudent[sortBy]) - +(prevStudent[sortBy]))
```
Comment (OleksiiBlezniuk) [inline src/sortStudents.ts:43]:
```suggestion
        return (Number(currtStudent[sortBy]) - Number(prevStudent[sortBy]))
```

Comment (OleksiiBlezniuk) [reply src/sortStudents.ts:43]: Number constructor is more readable

File: src/sortStudents.ts [hunk 6]
```diff
@@ -1,16 +1,55 @@
...
 
+      case SortType.AverageGrade:
+        return (calculateAverageGrade(currtStudent)
+        - calculateAverageGrade(prevStudent))
+          * orderCondition;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+      default:
+        throw new Error(`Sort type ${sortBy} is invalid`);
```
Comment (OleksiiBlezniuk) [inline src/sortStudents.ts:52]: Error is overkill here, it's enough to return 0 here

**Review by OleksiiBlezniuk — APPROVED** [review]

**Pull request closed without merging**